spec/unit/module/git_spec.rb

```diff
@@ -268,6 +268,48 @@ def test_module(extra_opts, env=nil)
             end
           end
         end
+
+        context "when using default_branch_override" do
+          before(:each) do
+            allow(mock_repo).to receive(:resolve).with(mock_env.ref).and_return(nil)
+          end
+
+          context "and the default branch override is resolvable" do
+            it "uses the override" do
+              expect(mock_repo).to receive(:resolve).with('default_override').and_return('5566aabb')
+              mod = test_module({branch: :control_branch,
+                                 default_branch: 'default',
+                                 default_branch_override: 'default_override'},
+                                 mock_env)
+              expect(mod.properties).to include(expected: 'default_override')
+            end
+          end
+
+          context "when default branch override is not resolvable" do
+            context "and default branch is provided" do
```

Comment:
Should we test the case where the default branch is provided, but also not resolvable? That feels different from just not providing a default branch.

Reply:
We already test default suceeding (proving that we fall back), and elsewhere we test the default failing, demonstrating what that looks like. It feels like it's covered to me. But I can add a case if you think that's not sufficient.

Reply:
I'm not familiar with the tests in this repo, so if you think it's covered then I trust you.
I was just thinking that the default branch being unresolvable has different consequences depending on whether you use a default_override or not. When using a default_override, the default being unresolvable only matters if the default_override is also unresolvable.

Reply:
I added something for that last case. If it's not resolvable and the override is, that doesn't matter because we never try it.